File: server/test_rerank.py
import urllib.request, json, time

BASE = '<INTERNAL_URL_REMOVED>

def test(query, session_id):
    data = json.dumps({'message': query, 'sessionId': session_id}).encode()
    req = urllib.request.Request(BASE, data=data, headers={'Content-Type': 'application/json'})
    t0 = time.time()
    resp = urllib.request.urlopen(req, timeout=90)
    result = json.loads(resp.read().decode())
    elapsed = time.time() - t0
    return result, elapsed

print('=== RAG 优化测试（含 Rerank）===')
print()

# 测试1：短查询（触发查询改写）
print('[测试1] 短查询 + 查询改写')
result, t = test('报销', 'test-1')
print(f'  查询: 报销')
print(f'  回复: {result.get("reply", "")[:100]}')
print(f'  意图: {result.get("intent", "")}')
print(f'  耗时: {t:.2f}s')
print()

# 测试2：标准查询
print('[测试2] 标准查询')
result, t = test('如何申请费用报销', 'test-2')
print(f'  查询: 如何申请费用报销')
print(f'  回复: {result.get("reply", "")[:100]}')
print(f'  意图: {result.get("intent", "")}')
print(f'  耗时: {t:.2f}s')
print()

# 测试3：检查后端日志中的 Rerank 调用
print('[测试3] 检查 Rerank 是否生效')
print('  请查看后端控制台输出，应看到类似日志：')
print('  [Rerank] 查询: ...  文档数: N  耗时: Xs')
print()

print('=== 测试完成 ===')

File: server/test_rerank.js
const { rerankResults } = require('./vector-store.js');

async function test() {
  const query = '如何报销';
  const candidates = [
    { content: '报销流程说明', title: '报销流程' },
    { content: '付款申请流程', title: '付款申请' },
    { content: '费用报销管理制度', title: '费用报销' }
  ];
  
  console.log('开始测试 rerankResults 函数...');
  console.log('查询:', query);
  console.log('候选数:', candidates.length);
  
  try {
    const result = await rerankResults(query, candidates, 5);
    console.log('重排序结果:');
    result.forEach((r, i) => {
      console.log(`  ${i+1}. ${r.title} (score: ${r.rerankScore})`);
    });
    console.log('✅ 测试通过');
  } catch (e) {
    console.error('❌ 测试失败:', e.message);
    console.error(e.stack);
  }
}

test();
